File: CaseStudies/glass/Implementations/C/src/InputFormat.c
#include "InputFormat.h"

#include <stdio.h>
#include <stdlib.h>

void get_input(char* filename, InputParameters* params){
  FILE* infile;
  infile = fopen(filename, "r");
  fscanf(infile, "%lf", &params->a);
  fscanf(infile, "%lf", &params->b);
  fscanf(infile, "%lf", &params->t);
  fscanf(infile, "%d", &params->gt);
  fscanf(infile, "%lf", &params->w);
  fscanf(infile, "%lf", &params->tnt);
  fscanf(infile, "%lf", &params->sdx);
  fscanf(infile, "%lf", &params->sdy);
  fscanf(infile, "%lf", &params->sdz);
  fscanf(infile, "%lf", &params->pbtol);
  fclose(infile);
}

File: code/stable/glassbr/src/python/InputFormat.py
from __future__ import print_function
import sys
import math

import InputParameters

def get_input(filename, inParams):
    outfile = open("log.txt", "a")
    print("function get_input called with inputs: {", file=outfile)
    print("  filename = ", end='', file=outfile)
    print(filename, end='', file=outfile)
    print(", ", file=outfile)
    print("  inParams = ", end='', file=outfile)
    print("Instance of InputParameters object", file=outfile)
    print("  }", file=outfile)
    outfile.close()
    
    lines = []
    linetokens = []
    infile = open(filename, "r")
    infile.readline()
    inParams.a = float(infile.readline())
    infile.readline()
    inParams.b = float(infile.readline())
    infile.readline()
    inParams.w = float(infile.readline())
    infile.readline()
    inParams.P_btol = float(infile.readline())
    infile.readline()
    inParams.TNT = float(infile.readline())
    infile.readline()
    inParams.g = infile.readline().rstrip()
    infile.readline()
    inParams.t = float(infile.readline())
    infile.readline()
    inParams.SD_x = float(infile.readline())
    infile.readline()
    inParams.SD_y = float(infile.readline())
    infile.readline()
    inParams.SD_z = float(infile.readline())
    infile.close()


